addons/account_payment/models/payment_transaction.py

```diff
@@ -106,7 +107,12 @@ def _post_process(self):
         super()._post_process()
         for tx in self.filtered(lambda t: t.state == 'done'):
             # Validate invoices automatically once the transaction is confirmed.
-            self.invoice_ids.filtered(lambda inv: inv.state == 'draft').action_post()
+            try:
+                with self.env.cr.savepoint():
```

Comment:
I don't think that the savepoint is needed here. In particular, it should never be in a loop; see https://github.com/odoo/odoo/pull/167236